### ISSUE TITLE
gsl/benchmarks/blas/blas_klee_correct.x86_64 is incorrect

### ISSUE THREAD

**Issue by delcypher** [delcypher/fp-benchmarks-imperial#2]
This benchmark is supposed to be correct according to the `spec.yml` file but our tool found a bug in it (during testing we probably didn't let our tool run long enough to find the bug).

```
WARNING:MainThread: MISMATCH SPEC for task no_assert_fail:
/home/user/build_O2/benchmarks/c/imperial/gsl/benchmarks/blas_klee_correct.x86_64.bc (/home/dsl11/dev/klee-afr/experiments/first/wd/workdir-44/klee-wd)

WARNING:MainThread:   Test 000001 in /home/user/fp-bench/benchmarks/c/imperial/gsl/benchmarks/blas/blas-sym.c:46
```

The particular failure is

```
KLEE: ERROR: /home/user/fp-bench/benchmarks/c/imperial/gsl/benchmarks/blas/blas-sym.c:46: ASSERTION FAIL: c[0] == a[0]*b[0] + a[1]*b[2] + a[2]*b[4]
```

Examining the ktest file reveals that the problematic value is a `nan` which the test does not guard against.
We should probably add a `klee_assume(!isnan(a[0]))` or something?

@ccadar This is your benchmark. Any thoughts on it?

**Comment by ccadar** [comment]
You're right, I forgot to eliminate the NaN case.  I've just done this in https://gitlab.doc.ic.ac.uk/afd/fp-benchmarks/compare/5d47ebf03c9ce4537dcc7a393f9b653ef4cfd4ce...2a0d6f8dc047223ed84f710c10a62ede224b4d89.  Or have you moved the repo in the meantime?

**Comment by delcypher** [comment]
@ccadar You've pushed to the wrong repo. That's the old internal one. You need to push to https://github.com/delcypher/fp-benchmarks-imperial/

**Comment by ccadar** [comment]
I suspected that :)  Should we just remove that one to avoid confusion?

**Comment by ccadar** [comment]
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: At least one approved review is required
To https://github.com/delcypher/fp-benchmarks-imperial
 ! [remote rejected] master -> master (protected branch hook declined)
error: failed to push some refs to 'https://github.com/delcypher/fp-benchmarks-imperial'

**Comment by delcypher** [comment]
@ccadar The history was rewritten so you need to clone a fresh copy.

**Comment by delcypher** [comment]
@ccadar I don't want to dump the old repo just yet because it has useful history.

**Comment by ccadar** [comment]
OK about the old repo, although I thought you moved the history to GitHub

**Comment by ccadar** [comment]
Regarding my permission error, was your suggestion to clone a fresh copy?  That's what I did.

**Comment by delcypher** [comment]
@ccadar Oh I might have set up some protection to stop people from pushing to it without a review. Can you open a PR and I'll merge it.

**Comment by ccadar** [comment]
Ok.  I didn't know about this feature, but it's great, I'll enable it on other repos I own.

**Comment by delcypher** [comment]
@ccadar Yes it's a nice feature. There's also a feature to block force pushes which I highly recommend you switch on.

**Comment by delcypher** [comment]
@ccadar The feature is called "protected branches".

**Comment by ccadar** [comment]
Yeap, found it, thanks!

**Comment by delcypher** [comment]
I think this is fixed by #5. I will close this issue once I've confirmed this.

**Comment by delcypher** [comment]
klee-afr in my experiments did not give a conclusion within the timeout (30 minutes) for this benchmark so closing for now.
